packages/webapp/src/hooks/api/useServiceAnswerList/useAddServiceAnswerCallback.ts

```diff
@@ -53,15 +54,66 @@ export function useAddServiceAnswerCallback(refetch: () => void): AddServiceAnsw
 					})
 				}
 
-				await addServiceAnswers({ variables: { serviceAnswer } })
-				refetch()
+				// The service answer we will use for our optimistic response. Need to ensure value and values are populated
+				// when writing to the cache
+				const optimisticServiceAnswer = {
+					..._serviceAnswer,
+					fields: _serviceAnswer.fields.map((field) => {
+						const f = field
+
+						// Single field value
+						if (typeof field.value === 'undefined') f.value = null
+
+						// Multi field value
+						if (typeof field.values === 'undefined') f.values = null
+
+						return f
+					})
+				}
+
+				addServiceAnswers({
+					variables: { serviceAnswer },
+					optimisticResponse: {
+						createServiceAnswer: {
+							message: 'Success',
+							serviceAnswer: {
+								...optimisticServiceAnswer,
+								id: crypto.randomUUID(),
+								__typename: 'ServiceAnswer'
+							},
+							__typename: 'ServiceAnswerResponse'
+						}
+					},
+					update: (cache, result) => {
+						// optimisticResponse or serverResponse
+						const newServiceAnswer = result.data.createServiceAnswer.serviceAnswer
+
+						// Fetch all the activeEngagements
+						const queryOptions = {
+							query: GET_SERVICE_ANSWERS,
+							variables: { serviceId: serviceAnswer.serviceId }
+						}
+
+						// Now we combine the newEngagement we passed in earlier with the existing data
+						const addOptimisticResponse = (data) => {
+							if (data) {
+								return {
+									serviceAnswers: [...data.serviceAnswers, newServiceAnswer]
+								}
+							}
+						}
+
+						cache.updateQuery(queryOptions, addOptimisticResponse)
+						// refetch() // TODO: not sure what refetch() does
```

Comment:
You need to add back `refetch` it's been used as a parameter to the hook, it's `useLoadServiceAnswersCallback` pass trough from the `index.ts` of the folder.

It's a technique to update the queries after a mutations.

Reply:
We'll leave this as is for now until it can be investigated further.